src/tests/users.test.js

```diff
@@ -0,0 +1,143 @@
+      res.body.data.should.have.property('date_of_birth');
+      res.body.data.should.have.property('preferred_language');
+      res.body.data.should.have.property('preferred_currency');
+      res.body.data.should.have.property('residential_address');
+    });
+  });
+
+  describe('PATCH /profile', () => {
+    const profileDetails = {
+      first_name: 'Elijah',
+      last_name: 'Enuem-Udogu',
+      gender: 'Male',
+      date_of_birth: '1994-05-20',
+      preferred_language: 'English',
+      residential_address: 'Benin City, Nigeria',
+      preferred_currency: 'Nigerian Naira (NGN)',
+    };
+
+    it('should return 401 if there is no token in the header', async () => {
+      const res = await chai.request(app)
+        .patch(`${URL_PREFIX}/profile`)
+        .set('Content-Type', 'application/json')
+        .send(profileDetails);
+
+      res.should.have.status(401);
+      res.body.should.have.property('status').eql('error');
+      res.body.should.have.property('error').eql('Authentication failed, please login');
+    });
+
+    it('should return 400 if the token in the header is invalid', async () => {
+      const res = await chai.request(app)
+        .patch(`${URL_PREFIX}/profile`)
+        .set('Content-Type', 'application/json')
+        .set('token', 'lsdjlfsjdlkfjsd')
+        .send(profileDetails);
+
+      res.should.have.status(400);
+      res.body.should.have.property('status').eql('error');
+      res.body.should.have.property('error').eql('Access Denied, Invalid or Expired Token');
+    });
+
+    it('should update the user\'s profile successfully if all conditions are met', async () => {
+      const payload = Helper.pickFields(verifiedUser, ['id', 'is_admin']);
+      const token = await Helper.genToken(payload);
+
+      const res = await chai.request(app)
+        .patch(`${URL_PREFIX}/profile`)
+        .set('Content-Type', 'application/json')
+        .set('token', token)
+        .send(profileDetails);
+
+      res.should.have.status(200);
+      res.body.should.have.property('status').eql('success');
+      res.body.should.have.property('data');
+      res.body.data.should.be.a('object');
+      res.body.data.should.have.property('first_name').eql('Elijah');
+      res.body.data.should.have.property('last_name').eql('Enuem-Udogu');
+      res.body.data.should.have.property('email').eql('koppter.kom@gmail.com');
+      res.body.data.should.have.property('is_admin');
+      res.body.data.should.have.property('is_verified');
+      res.body.data.should.have.property('role');
+      res.body.data.should.have.property('gender').eql('Male');
+      res.body.data.should.have.property('date_of_birth');
+      res.body.data.should.have.property('preferred_language').eql('English');
+      res.body.data.should.have.property('preferred_currency').eql('Nigerian Naira (NGN)');
+      res.body.data.should.have.property('residential_address').eql('Benin City, Nigeria');
+    });
+  });
+});
```

Comment:
I believe we have `auth.test.js` file, please don't abstract yours to a separate file.

Reply:
Alright